### DISCUSSION TITLE
ST7735 1.8" LCD does not work drivers

### DISCUSSION THREAD

**Discussion by ehsanvas** [micropython/micropython#13372] [Libraries & Drivers]
I have bought 20 new ST7735 KMR-1.8 SPI. The appearance and pinout are the same as the ones I have used before but the screen stays white with old codes. the supplier in China sent me a piece of code in C. I rewrote the init function in the Micropython screen is now initiated but colors and positions are completely wrong. 
Can anyone help me with this?
the drivers I have tested:
https://github.com/boochow/MicroPython-ST7735/blob/master/ST7735.py
https://github.com/GuyCarver/MicroPython/blob/master/lib/ST7735.py
https://github.com/boochow/MicroPython-ST7735
pin out:
![image](https://github.com/micropython/micropython/assets/5195633/6d1012a4-aed2-4d89-80e0-d2b1b27235d2)

**Comment by peterhinch** [comment]
The ST7735 has many variants. Graphics chips can also be wired to the display in various ways, requiring firmware to match. It's therefore best to buy a display board with known support.

[This doc](https://github.com/peterhinch/micropython-nano-gui/blob/master/DRIVERS.md#31-drivers-for-st7735r) is for drivers for the ST7735r as used in Adafruit display boards.

**Reply by ehsanvas** [reply]
Thanks
this model works with initR(INITR_BLACKTAB); with Arduino adafruit library.
But I did not find any alternative for BLACKTAB in micropython

**Comment by shariltumin** [comment]
Perhaps this [video](https://www.youtube.com/watch?v=FGqRiPqg4Ds) will help you to the correct pin connection.
